### PR TITLE
Better URL for scraping metrics from kubelet in node e2e tests

### PR DESCRIPTION
**Pull request by dims**
Do a pre-emptive lookup ourselves and use a host interface if we run into trouble with DNS 

Prevent failures due to DNS lookup [for example that looks like this one](https://storage.googleapis.com/kubernetes-jenkins/logs/ci-cgroupv2-containerd-node-arm64-e2e-serial-ec2/1673717387939024896/artifacts/i-0efe4a00c3f485970/i-0efe4a00c3f485970-ginkgo.log):
```
  [FAILED] Timed out after 5.000s.
  The function passed to Eventually returned the following error:
      <*url.Error | 0x4000e92db0>: 
      Get "http://i-0efe4a00c3f485970:10255/metrics": dial tcp: lookup i-0efe4a00c3f485970 on 172.31.0.2:53: no such host
      {
          Op: "Get",
          URL: "http://i-0efe4a00c3f485970:10255/metrics",
          Err: <*net.OpError | 0x4000b6aaa0>{
              Op: "dial",
              Net: "tcp",
              Source: nil,
              Addr: nil,
              Err: <*net.DNSError | 0x40008665c0>{
                  Err: "no such host",
                  Name: "i-0efe4a00c3f485970",
                  Server: "172.31.0.2:53",
                  IsTimeout: false,
                  IsTemporary: false,
                  IsNotFound: true,
              },
          },
      }
```

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e_node/util.go [hunk 1]
```diff
@@ -623,3 +624,22 @@ func WaitForPodInitContainerToFail(ctx context.Context, c clientset.Interface, n
 		return false, nil
 	})
 }
+
+func nodeNameOrIP() string {
```
Comment (SergeyKanzhelev) [inline test/e2e_node/util.go:628]: there are other instances:

- `logKubeletLatencyMetrics` uses `fmt.Sprintf("%s:%d", framework.TestContext.NodeName, ports.KubeletReadOnlyPort), "/metrics")`
- `getPodFromStandaloneKubelet` uses `endpoint := fmt.Sprintf("http://127.0.0.1:%d/pods", ports.KubeletReadOnlyPort)`

Would it be more reliable to use localhost here as well? If you can align those two usages as well, would be great

Comment (dims) [reply test/e2e_node/util.go:628]: I am comfortable fixing `logKubeletLatencyMetrics` but want to leave the `getPodFromStandaloneKubelet` alone, until i see evidence that we need to fix that. ok?

Comment (dims) [reply test/e2e_node/util.go:628]: Done!

Comment (SergeyKanzhelev) [reply test/e2e_node/util.go:628]: Would switching to `localhost` everywhere be better than adding this name resolution logic?

Comment (dims) [reply test/e2e_node/util.go:628]: @SergeyKanzhelev one reason i did this here was to ensure that i am not adversely affecting existing tests that may have an assumption on the node name or its ip (and falling back on the interface ONLY if resolution fails). I am happy to revisit this again to switch everything to `localhost`

Comment (SergeyKanzhelev) [reply test/e2e_node/util.go:628]: /lgtm

Comment (aojea) [reply test/e2e_node/util.go:628]: I recommend localhost, if possible , too

**Pull request merged**